src/renderer.rs

```diff
@@ -14,51 +15,122 @@ pub fn page(
+                    }
+                    tbody {
+                        @if !is_root {
+                            @if let Some(parent) = page_parent {
+                                tr {
+                                    td colspan="3" {
+                                        span.root-chevron { (chevron_left()) }
+                                        a.root href=(parametrized_link(&parent, &sort_method, &sort_order, &color_scheme)) {
+                                            "Parent directory"
+                                        }
                                     }
                                 }
                             }
                         }
+                        @for entry in entries {
+                            (entry_row(entry, &sort_method, &sort_order, &color_scheme))
+                        }
+                    }
+                }
+                a.back href="#top" {
+                    (arrow_up())
+                }
+            }
+        }
+    }
+}
+
+/// Partial: color scheme selector
+fn color_scheme_selector(
+    sort_method: &Option<listing::SortingMethod>,
+    sort_order: &Option<listing::SortingOrder>,
+    active_color_scheme: &themes::ColorScheme,
+) -> Markup {
+    html! {
+        nav {
+            ul {
+                li {
+                    a.change-theme href="#" title="Change theme" {
+                        "Change theme..."
                     }
-                    @for entry in entries {
-                        (entry_row(entry))
+                    ul {
+                        @for color_scheme in themes::ColorScheme::get_color_schemes() {
+                            @if active_color_scheme.get_name() == color_scheme.get_name() {
+                                li.active {
+                                    (color_scheme_link(&sort_method, &sort_order, &color_scheme))
+                                }
+                            } @else {
+                                li {
+                                    (color_scheme_link(&sort_method, &sort_order, &color_scheme))
+                                }
+                            }
+                        }
                     }
                 }
             }
-            a.back href="#top" {
-                (arrow_up())
+        }
+    }
+}
+
+/// Partial: color scheme link
+fn color_scheme_link(
+    sort_method: &Option<listing::SortingMethod>,
+    sort_order: &Option<listing::SortingOrder>,
+    color_scheme: &themes::ColorScheme,
+) -> Markup {
+    let link = parametrized_link("", &sort_method, &sort_order, &color_scheme);
+    let title = format!("Switch to {} theme", color_scheme.get_name());
+
+    html! {
+        a href=(link) title=(title) {
+            (color_scheme.get_name())
+            " "
+            @if color_scheme.is_dark() {
+                "(dark)"
+            } @ else {
```

Comment:
Can you fix the spacing here and in other places to make it consistent? I think it should be `} @else {` as that seems to be what you've done in most places.